kafka-agent/src/main/java/io/strimzi/kafka/agent/KafkaAgent.java

```diff
@@ -208,8 +230,15 @@ private void touch(File file) throws IOException {
     public static void premain(String agentArgs) {
         int index = agentArgs.indexOf(':');
         if (index == -1) {
-            LOGGER.error("Unable to parse arguments {}", agentArgs);
-            System.exit(1);
+            LOGGER.debug("Running in KRaft mode");
+            File brokerReadyFile = new File(agentArgs);
+            if (brokerReadyFile.exists() && !brokerReadyFile.delete()) {
+                LOGGER.error("Broker readiness file already exists and could not be deleted: {}", brokerReadyFile);
+                System.exit(1);
+            } else {
+                LOGGER.info("Starting KafkaAgent with brokerReadyFile={}", brokerReadyFile);
+                new KafkaAgent(brokerReadyFile).run();
+            }
         } else {
             File brokerReadyFile = new File(agentArgs.substring(0, index));
```

Comment:
Should we add a debug log here to show `Running in ZK mode` ?

Reply:
Yes, added